src/rpp/rpp/operators/flat_map.hpp

```diff
@@ -22,8 +22,8 @@ struct flat_map_t
 {
     RPP_NO_UNIQUE_ADDRESS Fn m_fn;
 
-    template<rpp::constraint::observable TObservable>
-        requires std::invocable<Fn, rpp::utils::extract_observable_type_t<TObservable>>
+    template<rpp::constraint::observable TObservable, typename ValueType = rpp::utils::extract_observable_type_t<TObservable>>
```

Comment:
I see your attempt with ValueType to avoid duplicates, but it would affect final binaries due to it would be compiled-in as template function with 2 params instead of one =C it is better to create special concept instead

Reply:
done